### DISCUSSION TITLE
Costs variable input

### DISCUSSION THREAD

**Discussion by mahewashabdi** [google/lightweight_mmm#15] [General]
Hi Team

I am trying to run this model on my input data and facing some confusion regarding the input Costs (Costs: The average cost per media unit per channel). Is it the CPM per channel. can you please elaborate this or how to perform calculation to get this. As this is one value per channel. I am using 4 channels so I should have 4 values per channels. I have cost spend and impressions per channel for each day.
Also I am using daily data, will that be fine for the model as the demo shows weekly data.

Thanks
Mahewash

**Comment by pabloduque0** [comment]
Hello @mahewashabdi ! Thanks for opening this discussion!
That one is a type and should say `Costs: The total cost per media unit per channel`, I have a pending commit to fix this which will push soon.

What it means is the total costs of each media channel over the whole period of time of the data. So you can obtain it by just summing the cost of each channel.

| Time | Channel 1 cost | Channel 2 cost |
| :----: | :----: | :----: |
| w1 | 14 | 15 |
| w2 | 89| 25 |

Cost for channel 1: 14 + 89
Cost for channel 2: 15 + 25

If you have a JAX/Numpy matrix you can just do: `jnp.sum(data, axis=0)` or numpy's equivalent (assuming axis 0 is the time dimmension). You can also do it with `pandas`.

Let me know if you have any other questions.

**Reply by pabloduque0** [reply]
Its fine to use daily data, and for costs nothing changes.

Just remember to add the `weekday_seasonality` in the fit method when training the model: `mmm.fit(..., weekday_seasonality=True)`

**Reply by mahewashabdi** [reply]
Thanks a lot for the prompt reply.

**Reply by mahewashabdi** [reply]
Hi @pabloduque0 

I am working on the optimization part. How to provide the prices ?

# Run media optimization.
budget = 40 # your budget here

here it looks like prices are scaled down and in the exampl demo I see Np.ones are used.

prices = np.array([0.1, 0.11, 0.12])
extra_features_test = extra_features_scaler.transform(extra_features_test)
solution = optimize_media.find_optimal_budgets(
    n_time_periods=extra_features_test.shape[0],
    media_mix_model=mmm,
    budget=budget,
    extra_features=extra_features_test,
    prices=prices)


I am not able to get this line '
Prices are the average price you would expect for the media units of each channel. If your data is already a money unit (eg. $) your prices should be an array of 1s.'
media data that I am using is the impressions per channel and cost has been calculated like you mentioned earlier.

![image](https://user-images.githubusercontent.com/63654651/173603724-cee4d542-ca50-4b3c-9dbf-e771ac442c4a.png)

Are these two different things ? how should I use the prices in my case.


Thanks 
Mahewash
